### PR TITLE
Feature/ab struct pred 231009

### PR DESCRIPTION
**Pull request by KirkD-Glysade**
Changes associated with Antibody Structure Prediction DF.

Parallel with same named branch in DataFxnPylib, DataFxns, and GlysadePython_mAb

### CODE REVIEW

File: test_df/test_mab_data_functions.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+def run_script(in_file: str, execute: Callable[[DataFunctionRequest], DataFunctionResponse]) -> Tuple[
+    DataFunctionRequest, DataFunctionResponse]:
+    request = request_from_file(in_file)
+    response = execute(request)
+    return request, response
+
+
+class mAb_DataFunctionTest(TestCase):
+
+    def test_antibody_structure_prediction(self) -> None:
```
Comment (jones-gareth) [inline test_df/test_mab_data_functions.py]: Similar to DataFxnPylib, we need to skip the test if we can't do the immune builder import (so the test suite can run on GlysadePython).  So try an import and if it fails print a warning message that the test is being skipped.

**Review by jones-gareth — COMMENTED** [review]
Looks good.  Does the mAB GlysadePython pass all the tests?  I guess it does not pass the chemistry tests.  
Maybe we will need to add a script to run all the mAB GlysadePython tests and a second script to run the GlysadePython tests.

**Comment by KirkD-Glysade** [comment]
> Looks good. Does the mAB GlysadePython pass all the tests? I guess it does not pass the chemistry tests. Maybe we will need to add a script to run all the mAB GlysadePython tests and a second script to run the GlysadePython tests.

This is why I created a different class for the mAb tests.  Almost all the tests in GlysadePython will fail in GlysadePython_mAb (except ANARCI tests, since ANARCI is installed in mAb, too).  Likewise, all the ImmuneBuilder tests will fail in GlysadePython.  The separate class only has tests for ImmuneBuilder and in PyCharm you have to set the right Python interpreter, etc.  Testing of the two seems like it has to be independent for each.

**Comment by jones-gareth** [comment]
> > Looks good. Does the mAB GlysadePython pass all the tests? I guess it does not pass the chemistry tests. Maybe we will need to add a script to run all the mAB GlysadePython tests and a second script to run the GlysadePython tests.
> 
> This is why I created a different class for the mAb tests. Almost all the tests in GlysadePython will fail in GlysadePython_mAb (except ANARCI tests, since ANARCI is installed in mAb, too). Likewise, all the ImmuneBuilder tests will fail in GlysadePython. The separate class only has tests for ImmuneBuilder and in PyCharm you have to set the right Python interpreter, etc. Testing of the two seems like it has to be independent for each.

Makes sense.  So unit test discovery (both unittest and nose) is done per directory. So I think all the mab tests need to go in a separate test_mab (or something similar) directory. Then to test mAb GlysadePython we do

````
cd test_mab
python3 -m unittest discover
````

And to test the regular python we could do (probably with a loop!)

```
cd test_bio
python3 -m unittest discover
cd ..
cd test_chem
python3 -m unittest discover
etc
```

The other possibility is to move `test_bio`, `test_chem`, `test_df` and `test_util` to a new 'test_glysade_python` folder, but it seems like we'd have to to a lot of refactoring (or keep changing PYTHONPATH).

**Comment by KirkD-Glysade** [comment]
I'll retool the mAb tests like you described.  It is small and easy to refactor at this point.  👍

**Comment by KirkD-Glysade** [comment]
Seemed to be as simple as creating a new directory an moving the various pieces - worked in my hands.  😉  Let me know if I've missed something.

**Comment by jones-gareth** [comment]
LGTM

**Pull request merged**